### PR TITLE
HC43 -  Upgrade diagnose functionality

### PR DESCRIPTION
**Pull request by Yarikks**

### CODE REVIEW

File: frontend/src/components/profile/common/constants/default-diagnosis-value.constant.ts [hunk 1]
```diff
@@ -1,7 +1,13 @@
...
+import { IDiagnosis } from 'common/interfaces';
 
-const DEFAULT_DIAGNOSIS_VALUE = {
+const DEFAULT_DIAGNOSIS_VALUE: IDiagnosis = {
+  [DiagnosisKey.ID]: '',
   [DiagnosisKey.DIAGNOSIS]: '',
+  [DiagnosisKey.DESCRIPTION]: '',
+  [DiagnosisKey.USER_ID]: '',
+  [DiagnosisKey.CREATED_AT]: '',
+  [DiagnosisKey.UPDATED_AT]: '',
```
Comment (ghost) [inline frontend/src/components/profile/common/constants/default-diagnosis-value.constant.ts]:
```suggestion
const DEFAULT_DIAGNOSIS_VALUE: ICreatedDiagnosis = {
  [DiagnosisKey.DIAGNOSIS]: '',
  [DiagnosisKey.DESCRIPTION]: '',
  [DiagnosisKey.USER_ID]: '',
```

File: frontend/src/components/profile/components/add-diagnosis-popup/add-diagnosis-popup.tsx [hunk 2]
```diff
@@ -0,0 +1,97 @@
...
+  InputColor,
+  InputType,
+} from 'common/enums';
+
+import styles from './add-diagnosis-popup.module.scss';
+
+type Props = {
+  userId: string;
+  isOpen: boolean;
+  onClose: () => void;
```
Comment (ghost) [inline frontend/src/components/profile/components/add-diagnosis-popup/add-diagnosis-popup.tsx]:
```suggestion
  onClose: BindingAction;
```

File: frontend/src/components/profile/components/add-diagnosis-popup/add-diagnosis-popup.tsx [hunk 3]
```diff
@@ -0,0 +1,97 @@
...
+};
+
+const AddDiagnosisPopup: React.FC<Props> = ({ userId, isOpen, onClose }) => {
+  const dispatch = useDispatch();
+
+  const { handleSubmit, reset, control, errors } = useForm<IDiagnosis>({
```
Comment (ghost) [inline frontend/src/components/profile/components/add-diagnosis-popup/add-diagnosis-popup.tsx]:
```suggestion
  const { handleSubmit, reset, control, errors } = useForm<ICreatedDiagnosis>({
```

File: frontend/src/components/profile/components/user-info/user-info.tsx [hunk 4]
```diff
@@ -52,6 +55,21 @@ const UserInfo: React.FC<Props> = ({ user, onEdit }) => {
...
+              : {
+                id: 'nerwksfd',
+                imagePath: '',
+                status: DocumentStatus.IN_REVIEW,
+                createdAt: '',
+                updatedAt: '',
+              }
+          }
+        />
+      )}
```
Comment (ghost) [inline frontend/src/components/profile/components/user-info/user-info.tsx]:
```suggestion
      {user.doctor.document && (
        <Documents
          document={document}
        />
      )}
```

File: frontend/src/services/diagnosis/diagnosis.service.ts [hunk 5]
```diff
@@ -21,20 +16,20 @@ class Diagnosis {
...
+    return this.#http.load(`${this.#apiPrefix}${ApiPath.DIAGNOSES}`, {
+      method: HttpMethod.GET,
+    });
   }
 
-  public create(userId: string, diagnosis: string): Promise<IDiagnosis> {
+  public create(
+    userId: string,
+    diagnosis: string,
+    description: string,
```
Comment (ghost) [inline frontend/src/services/diagnosis/diagnosis.service.ts]:
```suggestion
  createdDiagisis: ICreatedDiagnosis
```

**Review by ghost — CHANGES_REQUESTED** [review]

File: frontend/src/components/profile/components/user-info/user-info.tsx (hunk 4 above)
Comment (Yarikks) [reply frontend/src/components/profile/components/user-info/user-info.tsx]: oops, forgot to delete -_-

Comment (ghost) [reply frontend/src/components/profile/components/user-info/user-info.tsx]: Merge with dev. It's already fixed there

File: frontend/src/components/clinics/components/clinic/clinic.tsx [hunk 6]
```diff
@@ -13,10 +14,11 @@ const Clinic: React.FC<Props> = ({ clinic }) => {
   return (
     <Card
       title={clinic?.name}
-      label={clinic?.clinicType}
+      label={ClinicTypeToReadable[clinic?.clinicType]}
       btnLabel="More details"
       btnHref={AppRoute.CLINIC}
       imagePath={clinic?.imagePath}
+      alt={clinic?.name}
```
Comment (ghost) [inline frontend/src/components/clinics/components/clinic/clinic.tsx]:
```suggestion
```
Use title as `alt`

File: frontend/src/components/common/card/card.tsx [hunk 7]
```diff
@@ -11,6 +11,7 @@ type Props = {
   btnLabel?: string;
   btnHref?: AppRoute;
   imagePath: string;
+  alt: string;
```
Comment (ghost) [inline frontend/src/components/common/card/card.tsx]:
```suggestion
```

File: frontend/src/components/common/card/card.tsx [hunk 8]
```diff
@@ -21,18 +22,19 @@ const Card: React.FC<Props> = ({
   btnHref,
   imagePath,
   children,
+  alt,
 }) => {
   return (
     <div className={styles.cardContainer}>
       <div className={styles.imageContainer}>
-        <img className={styles.image} src={imagePath} />
+        <img className={styles.image} src={imagePath} alt={alt} />
```
Comment (ghost) [inline frontend/src/components/common/card/card.tsx]: Add width, height, loading attrs

File: frontend/src/components/doctors/components/doctor-item/doctors-item.tsx [hunk 9]
```diff
@@ -15,6 +15,7 @@ const DoctorItem: React.FC<Props> = ({ user }) => {
       label={user.doctor?.clinic?.clinicType}
       btnLabel="Make an appointment"
       imagePath={user.imagePath}
+      alt={user.name}
```
Comment (ghost) [inline frontend/src/components/doctors/components/doctor-item/doctors-item.tsx]:
```suggestion
```

**Review by ghost — CHANGES_REQUESTED** [review]

**Review by ghost — APPROVED** [review]
👍

**Pull request merged**